src/main/java/com/iotiq/user/security/AuthenticationController.java

```diff
@@ -93,4 +97,18 @@ private <T extends GrantedAuthority> PreAuthenticatedAuthenticationToken createP
         return new PreAuthenticatedAuthenticationToken(authenticationToken.getPrincipal(),
                 authenticationToken.getCredentials(), authorities);
     }
+
+    @GetMapping("/me")
+    public UserDto getCurrentUser(Principal principal) {
+        if (principal == null) throw new InvalidCredentialException();
+        User user = userService.find(principal.getName());
```

Comment:
userService contains different find functions byId and byName. May be should be better rename as findByUserName and findById.

Reply:
It has been changed.